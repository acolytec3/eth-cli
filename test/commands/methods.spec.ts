import MethodsCommand from '../../src/commands/methods'

describe('methods', () => {
  let stdoutResult: any

  beforeEach(() => {
    stdoutResult = []
    jest
      .spyOn(process.stdout, 'write')
      .mockImplementation(val => stdoutResult.push(require('strip-ansi')(val.toString())))
  })

  afterEach(() => jest.restoreAllMocks())

  it(`Should run 'methods' with empty args and flags and throw an error.`, async () => {
    await expect(MethodsCommand.run()).rejects.toThrow()
  })

  it(`Should run 'methods' and throw an error.`, async () => {
    await expect(MethodsCommand.run([])).rejects.toThrow(
      'Missing 1 required arg:\n' + 'abi  Contract ABI.\n' + 'See more help with --help',
    )
  })

  it(`Should run 'methods --help' and throw an error.`, async () => {
    await expect(MethodsCommand.run(['--help'])).rejects.toThrow('EEXIT: 0')
  })

  it(`Should run 'methods --abi' and throw an error.`, async () => {
    await expect(MethodsCommand.run(['LOL'])).rejects.toThrow(
      `ENOENT: no such file or directory, open 'LOL'`,
    )
  })

  it(`Should run 'methods ./test/files/contracts/Proxy.abi' and success.`, async () => {
    await MethodsCommand.run(['./test/files/contracts/Proxy.abi'])

    expect(stdoutResult).toMatchSnapshot()
  })
})
