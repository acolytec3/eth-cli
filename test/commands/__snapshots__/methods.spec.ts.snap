// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`methods Should run 'methods ./test/files/contracts/Proxy.abi' and success. 1`] = `
Array [
  "06fdde03	name()
",
  "085d4883	provider()
",
  "095ea7b3	approve(address,uint256)
",
  "18160ddd	totalSupply()
",
  "23b872dd	transferFrom(address,address,uint256)
",
  "2cd18565	proxy_admin()
",
  "2d37e6dc	getCrowdsaleInfo()
",
  "313ce567	decimals()
",
  "40556191	getCrowdsaleWhitelist()
",
  "4bc091a3	getCrowdsaleStartAndEndTimes()
",
  "56f7cafe	registry_exec_id()
",
  "5cad7cfb	getCrowdsaleStatus()
",
  "66188463	decreaseApproval(address,uint256)
",
  "6a73820e	app_name()
",
  "6e9960c3	getAdmin()
",
  "70a08231	balanceOf(address)
",
  "792651b0	app_storage()
",
  "95d89b41	symbol()
",
  "a6f2ae3a	buy()
",
  "a9059cbb	transfer(address,uint256)
",
  "d5d09021	isCrowdsaleFull()
",
  "d73dd623	increaseApproval(address,uint256)
",
  "d76ecb87	app_exec_id()
",
  "d9ba32fc	getWhitelistStatus(address)
",
  "d9cb8735	app_index()
",
  "db2e1741	app_version()
",
  "dd62ed3e	allowance(address,address)
",
  "ee7c0db0	getTokensSold()
",
  "f716c400	getCrowdsaleUniqueBuyers()
",
  "cfcd8c2d	init(address,uint256,uint256,uint256,uint256,uint256,uint256,bool,address,bool)
",
  "7185393c	updateAppExec(address)
",
  "3e7f54a9	updateAppInstance()
",
  "55f86501	exec(bytes)
",
]
`;
